controllers/helmrelease_controller_chart.go

```diff
@@ -98,7 +98,16 @@ func (r *HelmReleaseReconciler) loadHelmChart(source *sourcev1.HelmChart) (*char
 	defer f.Close()
 	defer os.Remove(f.Name())
 
-	res, err := http.Get(source.GetArtifact().URL)
+	url := source.GetArtifact().URL
+	if hostname := os.Getenv("SOURCE_CONTROLLER_LOCALHOST"); hostname != "" {
+		url = fmt.Sprintf("http://%s/%s/%s/%s/latest.tar.gz",
```

Comment:
Shouldn’t this address simply be advertised by the resource object if is running locally?

Reply:
The whole point of this is that you can run only HC locally.

Reply:
If someone whats to contribute to HC, now it needs to compile and run SC locally and that bings lots of pain due to libgit2. With this, you can debug HC locally while all the others controllers run in-cluster. Please see the guide I wrote in this PR.

Reply:
Check, I think it would be better to mutate the URL by parsing it using `url.Parse` and then modify the hostname. As `latest.tar.gz` (and the rest of the URL format) is an assumption that does not work for charts: https://github.com/fluxcd/source-controller/blob/568b2845c7e06abdb0c4efb365abf91a819225d5/controllers/helmchart_controller.go#L444

Reply:
Ah I forgot about that...

Reply:
That's what reviews are for :-D